### ISSUE TITLE
Error: Debug Failure. False expression. in 'getNodes' during 'getCodeFixes'

### ISSUE THREAD

**Issue by aozgaa** [Microsoft/TypeScript#22560]
This issue comes from crash dumps in telemetry. We've tried to de-duplicate issues on a best-effort basis, comparing the sequence of methods called and the command requested while ignoring line numbers.
**TypeScript Version Prefix**: 2.7.2
**VSCode Version** 1.20.1
**command requested**: getCodeFixes
**hitting sessions**: 106
**proportion of all sessions**: 0.007555238774055595
**five line hash**: 9cab4031d5846e63579d64d946770cd1
**stack**
```
Error: Debug Failure. False expression.
    at getNodes (tsserver.js:77758:22)
    at Object.getCodeActions (tsserver.js:77730:29)
    at unknown (tsserver.js:77219:33)
    at Object.forEach (tsserver.js:1431:30)
    at Object.getFixes (tsserver.js:77218:16)
    at unknown (tsserver.js:83366:35)
    at Object.flatMap (tsserver.js:1698:25)
    at Object.getCodeFixesAtPosition (tsserver.js:83364:23)
    at IOSession.Session.getCodeFixes (tsserver.js:90388:64)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (tsserver.js:89292:61)
    at unknown (tsserver.js:90553:88)
    at IOSession.Session.executeWithRequestId (tsserver.js:90544:28)
    at IOSession.Session.executeCommand (tsserver.js:90553:33)
    at IOSession.Session.onMessage (tsserver.js:90573:35)
    at Interface.<anonymous> (tsserver.js:91774:27)
    at emitOne (events.js:96:13)
    at Interface.emit (events.js:191:7)
    at Interface._onLine (readline.js:241:10)
    at Interface._normalWrite (readline.js:384:12)
    at Socket.ondata (readline.js:101:10)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:191:7)
    at readableAddChunk (_stream_readable.js:178:18)
    at Socket.Readable.push (_stream_readable.js:136:10)
    at Pipe.onread (net.js:560:20)
```

**Comment by andy-ms** [comment]
Already fixed in master branch by  #21886 -- now `if (token.kind !== SyntaxKind.ThisKeyword) return undefined;` instead of an assert.
Duplicate of #21812 -- @aozgaa Any idea why this wasn't disambiguated like it was last time?

**Comment by aozgaa** [comment]
When considering an incoming stack, I changed the method of checking for existing issues.
Before, we scanned through all of the issues and comparing the first 5 non-message lines of the stack.
Now, I've added a hash of the first five non-message lines of each stack to the body of the issue, so we can use github's search to find matching issues. It looks like the digests of past issues don't agree, however. I'll look into understanding/fixing this discrepancy ASAP.

Thanks for catching this.
